docs/grid-introducer.rst

```diff
@@ -0,0 +1,104 @@
+You want your Tahoe-LAFS client to be able to find storage servers to access.
+This means your client needs both current and future connection information for those storage servers.
+By configuring your client with one small static connection hint,
+the "Grid Introducer" system enables the client to find more information and future updates to that information.
+
+Before attempting this configuration,
+you should have the Grid Introducer *cap* and *furl* which you received from the operator of a Tahoe-LAFS storage grid which uses the Grid Introducer system.
+
+The Grid Introducer is configured in ``tahoe.cfg`` in the ``[client]`` section's ``grid-introducer.cap`` and ``grid-introducer.furl`` items.
+For example::
+
+  [client]
+  grid-introducer.cap = URI:DIR2-RO:4bnx...
+  grid-introducer.furl = pb://sokl...@192.168.69.247:44801/eqpw...
+
+Start a Tahoe-LAFS client node with these items configured and the client will be able to find and follow all storage servers that are part of that grid.
+
+Use As A Storage Provider
+-------------------------
+
+You want your Tahoe-LAFS storage node to be able to publish its connection details where clients can find it.
+Using Grid Introducer,
+a storage server will discover other storage servers and write its connection details to all of them.
+Then the capability to read this information is linked into a collection stored on the grid and known to clients.
+
+A storage server must be configured just like a storage client so it can discover other storage servers.
+See `Use As A Client`_ for this part of the configuration.
+
+Next, a storage server must be configured to maintain its announcement on the grid.
+This is also done in ``tahoe.cfg``,
+in the ``[storage]`` section's ``grid-introducer.enable`` item.
+
+For example::
+
+  [storage]
+  grid-introducer.enable = true
+
+After the storage server has been configured this way and started it will upload its announcement.
+
+Use As A Grid Coordinator
+-------------------------
+
+Setup
+~~~~~
+
+You want to offer a collection of storage servers as a Tahoe-LAFS storage grid.
+
+The ``grid-introducer`` tool requires a client node it can use to store and publish announcements.
+With that in place,
+the first step is to create the introducer's persistent state::
+
+  grid-introducer create --config <path> --client-api-root <Tahoe-LAFS client node HTTP API root>
+
+This is an error if ``<path>`` exists already.
+If it does not then a new grid introducer configuration is written there.
+If ``<path>`` is ``-`` then the configuration is written to stdout.
+
+This is the grid introducer's state and is required by future commands.
+It is the operator's responsibility to persist this state.
+
+Enrolling Servers
+~~~~~~~~~~~~~~~~~
+
+A Tahoe-LAFS storage server which is to be enrolled first shares its announcement readcap with you.
+Then, you will add it to the announcement directory::
+
+   grid-introducer add-storage-server --config <path> --readcap URI:CHK-RO:5cmy...
+
```

Comment:
Previously, it said these are indexed by public-key; where does it get that from? (And I guess if we have pet-names, this is where the grid-operator would assign one).

Reply:
The public key should be in the announcement, so it would go and read the announcement to get it (maybe also doing some basic checks to make sure you didn't supply a write cap by mistake, or the cap or some unrelated data, or copy/paste it wrong, etc).

If we link by petname, which I think is better, the petname can be a command line argument _or_ maybe we could have a `--use-announcement-nickname` option that makes it read the nickname from the announcement (because there is one) and use that.  Handy if one person actually controls both sides and wants to save a little typing or remove one chance to copy/paste the wrong string.  Maybe.  Maybe that's a follow-up feature (if this even gets implemented).